### PR TITLE
COL-1035 Add conversation posting to Canvas API client

### PR DESCRIPTION
**Pull request by paulkerschen**
https://jira.ets.berkeley.edu/jira/browse/COL-1035

### CODE REVIEW

File: node_modules/col-canvas/lib/api.js [hunk 1]
```diff
@@ -440,6 +440,54 @@ var deleteFile = module.exports.deleteFile = function(ctx, file, callback) {
...
+    'url': url,
+    'method': 'POST',
+    'headers': {
+      'Authorization': util.format('Bearer %s', ctx.course.canvas.api_key)
+    },
+    'body': {
+      'as_user_id': ctx.user.canvas_user_id,
+      'recipients': [ recipientId ],
+      'subject': subject,
+      'body': body,
```
Comment (johncrossman) [inline node_modules/col-canvas/lib/api.js]: @paulkerschen unnecessary comma above

File: node_modules/col-canvas/lib/api.js [hunk 2]
```diff
@@ -440,6 +440,54 @@ var deleteFile = module.exports.deleteFile = function(ctx, file, callback) {
...
+    'json': true
+  };
+  request(opts, function(err, response, body) {
+    if (err) {
+      log.error({'err': err, 'course': ctx.course.id, 'opts': opts}, 'Failed to post to a Canvas conversation');
+      return callback({'code': 500, 'msg': 'Failed to post to a Canvas conversation'});
+    } else if (response.statusCode !== 201) {
+      log.error({
+        'statusCode': response.statusCode,
+        'responseBody': response.bdody,
```
Comment (johncrossman) [inline node_modules/col-canvas/lib/api.js]: @paulkerschen I like "bdody".  Is that what the kids are saying or is this a typo?

Comment (paulkerschen) [inline node_modules/col-canvas/lib/api.js]: @johncrossman If you've gdot to ask...

**Comment by johncrossman** [comment]
Looks bgood, @paulkerschen  Waiting on the man.

**Pull request merged**